### PR TITLE
feat: SDA-2614 (Implement native electron notifications)

### PR DESCRIPTION
**Pull request by KiranNiranjan**
## Description
The feature that adds support for inline reply for notifications
[SDA-2614](https://perzoinc.atlassian.net/browse/SDA-2614)
[SDA-2615](https://perzoinc.atlassian.net/browse/SDA-2615)
[SDA-2618](https://perzoinc.atlassian.net/browse/SDA-2618)

## Solution Approach
- Use Electron notification to display a notification
- Create a helper to choose between custom/native notification
- Bump SDA API version from `2.0.0` to `3.0.0`
- Remove remote module usage

## Related PRs
List related PRs against other branches/repositories:

branch | PR
------ | ------
SFE-Lite | [link](https://github.com/SymphonyOSF/SFE-Lite/pull/4993)

Notes: Add support for native electron notifications

### CODE REVIEW

File: src/app/notifications/notification-helper.ts [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+     */
+    public async closeNotification(id: number) {
+        if (this.electronNotification.has(id)) {
+            const electronNotification = this.electronNotification.get(id);
+            if (electronNotification) {
+                electronNotification.close();
+            }
+        } else {
+            await notification.hideNotification(id);
+        }
```
Comment (VishwasShashidhar) [inline src/app/notifications/notification-helper.ts:46]:
```suggestion
        if (this.electronNotification.has(id)) {
            const electronNotification = this.electronNotification.get(id);
            if (electronNotification) {
                electronNotification.close();
            }
            return;
        }
        await notification.hideNotification(id);
```

File: src/app/notifications/notification-helper.ts [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+    }
+
+    /**
+     * Method that sends the notification actions event
+     * on to web client
```
Comment (VishwasShashidhar) [inline src/app/notifications/notification-helper.ts:51]:
```suggestion
    /**
     * Sends the notification actions event to web client
```

File: src/app/notifications/notification-helper.ts [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+        extraData: ElectronNotificationData,
+    ) {
+        const mainWindow = windowHandler.getMainWindow();
+        if (mainWindow && windowExists(mainWindow) && mainWindow.webContents) {
+            mainWindow.webContents.send('notification-actions', { event, data, extraData });
+        }
+    }
+
+    private getIcon(options: INotificationData): string | undefined {
+        return isWindowsOS ? options.icon : undefined;
```
Comment (VishwasShashidhar) [inline src/app/notifications/notification-helper.ts:69]: What happens for macOS and Linux? Will those operating systems take app's default icon?

File: src/renderer/preload-main.ts [hunk 4]
```diff
@@ -22,7 +22,6 @@ const minMemoryFetchInterval = 4 * 60 * 60 * 1000;
 const maxMemoryFetchInterval = 12 * 60 * 60 * 1000;
 const snackBar = new SnackBar();
 const banner = new MessageBanner();
-const notification = remote.require('../renderer/notification').notification;
```
Comment (VishwasShashidhar) [inline src/renderer/preload-main.ts:25]: Excellent! The less we use remote, the better.

**Review by VishwasShashidhar — CHANGES_REQUESTED** [review]
@KiranNiranjan Apart from a few minor changes, looks good.

**Comment by VishwasShashidhar** [comment]
@KiranNiranjan Looks like the unit tests are failing...

File: src/app/notifications/notification-helper.ts (hunk 3 above)
Comment (KiranNiranjan) [reply src/app/notifications/notification-helper.ts:69]: Yes, That's correct

**Review by psjostrom — APPROVED** [review]
Looks very good!

**Review by VishwasShashidhar — APPROVED** [review]

**Review by keerthi16 — APPROVED** [review]

**Pull request merged**